pkg/lightning/backend/local/local.go

```diff
@@ -902,11 +903,19 @@ func splitRangeBySizeProps(fullRange Range, sizeProps *sizeProperties, sizeLimit
 	curKeys := uint64(0)
 	curKey := fullRange.start
 	sizeProps.iter(func(p *rangeProperty) bool {
+		if bytes.Equal(p.Key, engineMetaKey) {
+			return true
+		}
 		curSize += p.Size
 		curKeys += p.Keys
 		if int64(curSize) >= sizeLimit || int64(curKeys) >= keysLimit {
-			ranges = append(ranges, Range{start: curKey, end: p.Key})
-			curKey = p.Key
+			// in case the sizeLimit or keysLimit is too small
+			endKey := p.Key
+			if bytes.Equal(curKey, endKey) {
+				endKey = nextKey(endKey)
```

Comment:
could we just return

Reply:
If the region-split-size is set to a very small value that one kv can exceed this threshold, then the logic can exact cover this threshold, though such small threshold threshold is only used in tests

Reply:
I see, `return` will cause the range merged into next one.